doc/changes/changes-4.10.xml

```diff
@@ -0,0 +1,312 @@
+infinite groups in the filter <C>IsPGroup</C>.
+
+In GAP 4.10, we adjusted (in <URL><LinkText>&Hash;1545</LinkText><Link>https://github.com/gap-system/gap/pull/1545</Link></URL>)
+the documentation for <C>IsPGroup</C> to the commonly accepted definition for
+p-groups. In addition, code in the GAP library and in packages using
+<C>IsPGroup</C> was audited and (in a very few cases) adjusted to explicitly check
+<C>IsFinite</C> (see e.g. <URL><LinkText>&Hash;2866</LinkText><Link>https://github.com/gap-system/gap/pull/2866</Link></URL>).
+</Item>
+
+<!-- 
+ -->
+<Mark>Experimental support for using the Julia garbage collector</Mark><Item>
+
+It is now possible to use the garbage collector of the <URL Text="Julia language">https://julialang.org</URL>
+instead of GAP's traditional GASMAN garbage collector. This is partly
+motivated by a desire to allow tight integration with GAP and Julia in the
+future. Warning: right now, this is <E>slower</E>, and also requires a patched
+version of Julia.
+<P/>
+
+Relevant pull requests:
+<URL><LinkText>&Hash;2092</LinkText><Link>https://github.com/gap-system/gap/pull/2092</Link></URL>,
+<URL><LinkText>&Hash;2408</LinkText><Link>https://github.com/gap-system/gap/pull/2408</Link></URL>,
+<URL><LinkText>&Hash;2461</LinkText><Link>https://github.com/gap-system/gap/pull/2461</Link></URL>,
+<URL><LinkText>&Hash;2485</LinkText><Link>https://github.com/gap-system/gap/pull/2485</Link></URL>,
+<URL><LinkText>&Hash;2495</LinkText><Link>https://github.com/gap-system/gap/pull/2495</Link></URL>,
+<URL><LinkText>&Hash;2672</LinkText><Link>https://github.com/gap-system/gap/pull/2672</Link></URL>,
+<URL><LinkText>&Hash;2688</LinkText><Link>https://github.com/gap-system/gap/pull/2688</Link></URL>,
+<URL><LinkText>&Hash;2793</LinkText><Link>https://github.com/gap-system/gap/pull/2793</Link></URL>,
+<URL><LinkText>&Hash;2904</LinkText><Link>https://github.com/gap-system/gap/pull/2904</Link></URL>,
+<URL><LinkText>&Hash;2905</LinkText><Link>https://github.com/gap-system/gap/pull/2905</Link></URL>,
+<URL><LinkText>&Hash;2931</LinkText><Link>https://github.com/gap-system/gap/pull/2931</Link></URL>.
+</Item>
+
+<!-- 
+ -->
+<Mark>libGAP (WIP)</Mark><Item>
+
+We now provide a semi-official way to allow 3rd party code to link GAP as a
+library; this is based on the <C>libGAP</C> code by
+<URL Text="SageMath">https://www.sagemath.org</URL>, but different. We hope that a future
+version of SageMath can drop its custom modifications for GAP and use this
+interface instead. If you are interested in this kind of interface, please
+get in touch with us to help us improve it.
+See also <URL Text="this email">https://mail.gap-system.org/pipermail/gap/2018-August/001123.html</URL>.
+<P/>
+
+Relevant pull requests:
+<List>
+<Item><URL><LinkText>&Hash;1690</LinkText><Link>https://github.com/gap-system/gap/pull/1690</Link></URL> Add a callback to <C>FuncJUMP_TO_CATCH</C></Item>
```

Comment:
Here and below: new line after the URL would make the source much more readable. I can review all the text in the browser since it does not display very long lines conveniently, so I am mostly stopping the review here for now.